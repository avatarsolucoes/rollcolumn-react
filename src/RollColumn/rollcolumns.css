.rollcolumns{
  box-sizing: border-box;
  display: block;
  position: relative;
  margin: 0 auto;
  padding: 0;
  width: 100%;
  max-width: 100% !important;
  height: 100%;
  border: 0;
  overflow: hidden;
}

.rollcolumns .content {
  box-sizing: border-box;
  position: relative;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: flex-start;
  align-content: flex-start;
  justify-content: space-between;
  width: 100%;
  /* max-width: 100%; */
  height: 100%;
  -o-transition: all ease-in-out 0.3s;
  -ms-transition: all ease-in-out 0.3s;
  -moz-transition: all ease-in-out 0.3s;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all cubic-bezier(0.79, 0.14, 0.15, 0.86) 0.3s;
  border: 0;
}

.rollcolumns .item {
  box-sizing: border-box;
  display: block;
  position: relative;
  /* width: 50%;
  max-width: 50%; */
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  border: 0;
  height: 100%;
  border: 0;
}

/* LABEL */
.rollcolumns .label {
  cursor: pointer;
}

.rollcolumns .label.full {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.rollcolumns .label.disabled {
  filter: grayscale(100%);
}

.rollcolumns .label.full:hover {
  background-color: rgba(0, 0, 0, 0.2);
}

/* INPUT CONTROL */
.rollcolumns .dinamiccolsCtrl {
  display: none;
  visibility: hidden;
}

@keyframes rcfadein {
  0% { opacity: 0;}
  100% { opacity: 1;}
}

@keyframes rcfadeout {
  0% { opacity: 1;}
  100% { opacity: 0;}
}
