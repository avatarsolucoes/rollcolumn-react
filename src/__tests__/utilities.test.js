import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import RollColumn, { Column, LabelColumn, gotoColumn } from '..';

function Test() {
  function goto(columnNumber) {
    gotoColumn('myName', columnNumber, true);
  }

  function change(obj, ord) {
    // eslint-disable-next-line no-console
    console.log(ord, obj);
  }

  return (
    <RollColumn name="myName" onColumnChange={change}>
      <Column order={1}>
        <p>Content 1</p>
        <LabelColumn to={2}>goto 2</LabelColumn>
        <button type="button" className="goto" onClick={() => goto(2)}>
          GOTO 2
        </button>
      </Column>
      <Column order={2}>
        <p>Content 1</p>
        <LabelColumn to={1}>back to 1</LabelColumn>
        <LabelColumn to={3}>back to 3</LabelColumn>

        <button type="button" className="goto" onClick={() => goto(1)}>
          GOTO 1
        </button>
      </Column>
      <Column order={3}>
        <p>Content 3</p>
        <LabelColumn to={1}>back to 1</LabelColumn>
        <LabelColumn to={2}>back to 2</LabelColumn>
      </Column>
    </RollColumn>
  );
}

describe('Test formats', () => {
  it('is truthy', () => {
    expect(RollColumn).toBeTruthy();
  });

  it('render RollColumn', () => {
    const wrapper = mount(<Test />);
    wrapper
      .find('.goto')
      .at(0)
      .simulate('click');
    wrapper.update();
    wrapper
      .find('.goto')
      .at(1)
      .simulate('click');
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('render renderer RollColumn', () => {
    const wrapper = {
      text: renderer.create(<Test />).toJSON(),
    };
    expect(wrapper).toMatchSnapshot();
  });

  it(`Test shallow `, () => {
    const wrapper = shallow(<Test />).html();
    expect(wrapper).toMatchSnapshot();
  });
});
