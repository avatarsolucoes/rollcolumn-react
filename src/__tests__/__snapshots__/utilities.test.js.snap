// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test formats Test shallow  1`] = `"<style></style><div id=\\"myName\\" class=\\"\\"><div class=\\"\\" style=\\"width:300%;max-width:300%\\"><div id=\\"rollcolumn-item-myName-1\\" class=\\"\\"><p>Content 1</p><label id=\\"rollcolumn-label-myName-2\\" for=\\"\\" class=\\"\\">goto 2</label><button type=\\"button\\" class=\\"goto\\">GOTO 2</button></div><div id=\\"rollcolumn-item-myName-2\\" class=\\"\\"><p>Content 1</p><label id=\\"rollcolumn-label-myName-1\\" for=\\"\\" class=\\"\\">back to 1</label><label id=\\"rollcolumn-label-myName-3\\" for=\\"\\" class=\\"\\">back to 3</label><button type=\\"button\\" class=\\"goto\\">GOTO 1</button></div><div id=\\"rollcolumn-item-myName-3\\" class=\\"\\"><p>Content 3</p><label id=\\"rollcolumn-label-myName-1\\" for=\\"\\" class=\\"\\">back to 1</label><label id=\\"rollcolumn-label-myName-2\\" for=\\"\\" class=\\"\\">back to 2</label></div></div></div>"`;

exports[`Test formats render RollColumn 1`] = `ReactWrapper {}`;

exports[`Test formats render renderer RollColumn 1`] = `
Object {
  "text": Array [
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />,
    <div
      className=""
      id="myName"
    >
      <div
        className=""
        style={
          Object {
            "maxWidth": "300%",
            "width": "300%",
          }
        }
      >
        <div
          className=""
          id="rollcolumn-item-myName-1"
          style={
            Object {
              "order": undefined,
            }
          }
        >
          <p>
            Content 1
          </p>
          <label
            className=""
            htmlFor=""
            icon={null}
            id="rollcolumn-label-myName-2"
            onClick={[Function]}
            store={null}
            style={null}
          >
            goto 2
          </label>
          <button
            className="goto"
            onClick={[Function]}
            type="button"
          >
            GOTO 2
          </button>
        </div>
        <div
          className=""
          id="rollcolumn-item-myName-2"
          style={
            Object {
              "order": undefined,
            }
          }
        >
          <p>
            Content 1
          </p>
          <label
            className=""
            htmlFor=""
            icon={null}
            id="rollcolumn-label-myName-1"
            onClick={[Function]}
            store={null}
            style={null}
          >
            back to 1
          </label>
          <label
            className=""
            htmlFor=""
            icon={null}
            id="rollcolumn-label-myName-3"
            onClick={[Function]}
            store={null}
            style={null}
          >
            back to 3
          </label>
          <button
            className="goto"
            onClick={[Function]}
            type="button"
          >
            GOTO 1
          </button>
        </div>
        <div
          className=""
          id="rollcolumn-item-myName-3"
          style={
            Object {
              "order": undefined,
            }
          }
        >
          <p>
            Content 3
          </p>
          <label
            className=""
            htmlFor=""
            icon={null}
            id="rollcolumn-label-myName-1"
            onClick={[Function]}
            store={null}
            style={null}
          >
            back to 1
          </label>
          <label
            className=""
            htmlFor=""
            icon={null}
            id="rollcolumn-label-myName-2"
            onClick={[Function]}
            store={null}
            style={null}
          >
            back to 2
          </label>
        </div>
      </div>
    </div>,
  ],
}
`;
